### PR TITLE
Reset the API HPA to previous values, 4min/8max scalability

### PR DESCRIPTION
**Pull request by jimleroyer**
## What are you changing?
- [ ] Releasing a new version of Notify
- [x] Changing kubernetes configuration

## Provide some background on the changes

We went through an incident this week where two clients hit the API hard for a short amount of time, possibly the root cause of bringing the API Kubernetes pod instances into an out of memory state. This caused restarts and a few messages to be dropped (around 20-25 observed in AWS CloudWatch / Logs Insights).

image

image

## If you are releasing a new version of Notify, what components are you updating
- [x] API
- [x] Admin
- [ ] Documentation
- [ ] Document download API
- [ ] Document download frontend

## Checklist if releasing new version:
- [ ] I made sure that the changes are as expected in [Notify staging](https://staging.notification.cdssandbox.xyz/)
- [ ] I have checked if the docker images I am referencing exist
    - [ ] [api lambda](https://ca-central-1.console.aws.amazon.com/ecr/repositories/private/296255494825/notify/api-lambda?region=ca-central-1) (requires Notification-Production / AdministratorAccess login)
    - [ ] [api k8s](https://gallery.ecr.aws/v6b8u5o6/notify-api)
    - [ ] [admin](https://gallery.ecr.aws/v6b8u5o6/notify-admin)
    - [ ] [documentation](https://gallery.ecr.aws/v6b8u5o6/notify-documentation)
    - [ ] [document download API](https://gallery.ecr.aws/v6b8u5o6/notify-document-download-api)
    - [ ] [document download frontend](https://gallery.ecr.aws/v6b8u5o6/notify-document-download-frontend)

## Checklist if making changes to Kubernetes:
- [ ] I know how to get kubectl credentials in case it catches on fire

## After merging this PR
- [ ] I have verified that the tests / deployment actions succeeded
- [ ] I have verified that any affected pods were restarted successfully
- [ ] I have verified that I can still log into [Notify production](https://notification.canada.ca)
- [ ] I have verified that the smoke tests still pass on production
- [ ] I have communicated the release in the #notify Slack channel.

### CODE REVIEW

File: env/production/replica_count.yaml [hunk 1]
```diff
@@ -72,4 +72,4 @@ metadata:
   namespace: notification-canada-ca
 spec:
   minReplicas: 1
-  maxReplicas: 1
+  maxReplicas: 2
```
Comment (sastels) [inline env/production/replica_count.yaml:75]: We probably don't need to ever have more than one `document-download-api` pod, but it doesn't hurt to have another available.

**Review by sastels — APPROVED** [review]
My only concern would be if it's possible to overload something else if the # pods get boosted (ie this would have killed the database back when api directly wrote new notifications). I can't think of anything likely though (ie we shouldn't get thousands of hits / minute of someone wanting updated status on a notification or something).

**Comment by jimleroyer** [comment]
We'll definitely need to keep an eye on applied pressure at key points in the data pipeline: API, SQS queues, workers, the database and Redis.

File: env/production/replica_count.yaml (hunk 1 above)
Comment (jimleroyer) [reply env/production/replica_count.yaml:75]: I increased it as I saw some errors in the past where the cause might have been insufficient resources. But I didn't investigate long enough to be certain of it.

**Pull request merged**